src/schematools/datasetcollection.py

```diff
@@ -18,27 +24,35 @@ class DatasetCollection(metaclass=Singleton):
     """
 
     def __init__(self) -> None:
-        self.datasets_cache = {}
-
-    def _load_dataset(self, dataset_id: str) -> Optional[DatasetSchema]:
-        """Loads the dataset from SCHEMA_URL.
+        """Initialize the DatasetCollection.
 
-        If SCHEMA_URL is not defined, return None.
+        Args:
+            schema_loader: An alternative schema loader can be provided
+                If schema_loader is None, the default url loader is used.
         """
-        # Avoid circular import problem.
-        from schematools.utils import dataset_schema_from_url
+        self.datasets_cache: Dict[str, DatasetSchema] = {}
+        self.schema_loader: Optional[loaders.SchemaLoader] = None
 
-        try:
-            schemas_url = os.environ["SCHEMA_URL"]
-        except KeyError:
-            return None
+    def set_schema_loader(self, schema_loader: loaders.SchemaLoader) -> None:
+        """Set the schema loader for the datasetcollection."""
+        self.schema_loader = schema_loader
+
+    def get_schema_loader(self) -> loaders.SchemaLoader:
+        """Get the schema_loader."""
+        if self.schema_loader is None:
+            raise ValueError("The datasetcollection should be initialized with a schema loader")
+        return self.schema_loader
 
-        return dataset_schema_from_url(schemas_url, dataset_id, prefetch_related=True)
+    def _load_dataset(self, dataset_id: str, prefetch_related: bool) -> Optional[DatasetSchema]:
+        """Loads the dataset, using the configured loader."""
+        schema_loader = cast(loaders.SchemaLoader, self.schema_loader)
```

Comment:
If you have `schema_loader` defined as an instance variable (see https://github.com/Amsterdam/schema-tools/pull/207/files#r749223188) You don't need the `cast` anymore. A simple `is not None` check would be sufficient in that case.